Packages/com.unity.inputsystem/CHANGELOG.md

```diff
@@ -9,9 +9,46 @@ however, it has to be formatted properly to pass verification tests.
 
 ## [Unreleased]
 
+### Changed
+
+- An upper limit of 1024 controls per device and 1kb of memory state per device has been introduced.
+  * This allows for certain optimizations.
+  * Should the limits prove too tight, they can be raised in the future.
+  * The most complex device we have at the moment (`Touchscreen`) has 242 controls and 616 bytes of state.
+
 ### Fixed
 
 - Fixed precompiled layouts such as `FastKeyboard` leading to build time regressions with il2cpp (case 1283676).
+- Fixed `InputUser.OnEvent` and `RebindingOperation.OnEvent` exhibiting bad performance profiles and leading to multi-millisecond input update times (case 1253371).
+  * In our own measurements, `InputUser.OnEvent` is >9 times faster than before and `RebindingOperation.OnEvent` is ~2.5 times faster.
+
+### Added
+
+- Added a new high-performance way to iterate over changed controls in an event.
+  ```CSharp
```

Comment:
You can store code samples in the project and then link to them. This way if they break in the future you will get a compile error. 
I just did it myself ;) https://github.cds.internal.unity3d.com/unity/com.unity.localization/pull/87

Reply:
Hmm, having to store examples in separate files seems like a major hassle. Half the time, I'd be too lazy to actually create an example. And then we'd have to ensure they get cleaned up, too, and stuff.

Wondering, though, whether we can instead create a test that just extracts all the examples and ensures they compile. Seems like a more fruitful avenue.